lib/spack/spack/test/packaging.py

```diff
@@ -85,6 +85,10 @@ def test_buildcache(mock_archive, tmpdir):
     with open(filename, "w") as script:
         script.write(spec.prefix)
 
+    # Create an absolute symlink
+    linkname = os.path.join(spec.prefix, "link_to_dummy.txt")
```

Comment:
I'm concerned this test isn't fully exercising the possible failures related to symlink relocation because the binary cache is installed in the same location where it was created from. Ideally the test (or a new test) would choose a new location to install to. Ideally the test would also confirm that the symlinks point to where you expect (right now it just does the binary package and install).

I think you may be able to avoid some of the setup that occurs here by just checking `binary_distribution.build_tarball`.

Reply:
Because the symlinks are changed into the workdir by `install_tree`, it's not possible to have the sort of hidden bug related to the final destination not changing that you're worried about here. 

If any of the symlinks are still pointing to the workdir after one install, the subsequent `create` command will fail. This test found a bug when I added linking to it through exactly that failure mode.

Reply:
I think your approach is sufficient for catching a number of bugs. Depending on the implementation though (or how it changes with future edits), I think it's possible that there are cases where unpacking into the same directory structure would succeed but unpacking into a different directory structure will fail. The test ensures that regardless of the implementation, the end desired result (successful relocation into a different root) succeeds.